### ISSUE TITLE
Difficulty appending steps / missing names

### ISSUE THREAD

**Issue by macartan** [DeclareDesign/DeclareDesign#259]
```
library(DesignLibrary)

processing <- declare_step(handler = fabricate,
                           neighbor = sapply(1:N, function(i) sum(Z[-i][G[-i]==G[i]])),
                           one_neighbor  = as.numeric(neighbor==1),
                           two_neighbors = as.numeric(neighbor==2))

linear     <- declare_estimator(Y ~ Z*neighbor, estimand = "Treat_1", model = lm_robust, 
                                label = "linear")

saturated  <- declare_estimator(Y ~ Z*one_neighbor + Z*two_neighbors, estimand = "Treat_1", 
                                label = "saturated", model = lm_robust)

neighbors_design <- insert_step(simple_spillover_designer(), after = 3, processing) 

# Issue 1: inserted step has no name
names(neighbors_design)

# Issue 2: append difficulties

check0 <- neighbors_design + linear 
check1 <- neighbors_design + saturated 
check2 <- neighbors_design + linear + saturated
```

1. inserted step did not create a name
2. check2 fails

```
> # Issue 1: inserted step has no name
> names(neighbors_design)
[1] "population" "estimand"   "assignment" ""           "reveal"     "estimator" 
> 
> # Issue 2: append difficulties
> 
> check0 <- neighbors_design + linear 
> check1 <- neighbors_design + saturated 
> check2 <- neighbors_design + linear   + saturated
Error in names(ret)[qnames != ""] <- qnames[qnames != ""] : 
  NAs are not allowed in subscripted assignmentsNAs are not allowed in subscripted assignments
```

**Comment by graemeblair** [comment]
yikes. the two are likely connected. will fix.

**Comment by graemeblair** [comment]
closed by #260
